packages/editor/src/components/post-publish-panel/maybe-tags-panel.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose, ifCondition } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { PanelBody } from '@wordpress/components';
+
+import FlatTermSelector from '../post-taxonomies/flat-term-selector';
+
+const TagsPanel = () => <PanelBody initialOpen={ false } title={ [
+	__( 'Tip:' ),
+	<span className="editor-post-publish-panel__link" key="label">{
+		__( 'Add tags to your post' )
+	}</span>,
+] }>
+	<p> { __( "Enter a few words that describe your post's subject to help interested readers find it." ) } </p>
+	<FlatTermSelector slug={ 'post_tag' } />
+</PanelBody>;
+
+class MaybeTagsPanel extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			hadTags: props.hasTags,
+		};
+	}
+
+	render() {
+		if ( this.state.hadTags ) {
```

Comment:
No, it can't be. We need to encode whether the post had tags _when the user clicked publish_ and the pre-publish panel was first shown. The `hasTags` can't be used for that because it will change to `true` if the user adds a new tag within the pre-publish panel. That will cause a re-render of this component and a new `hasTags` check: as a result the panel will be hidden preventing the user from adding more than one tag.

Reply:
Could `prevProps` help there to keep it visible?

Reply:
`prevProps` would be enough up to the second tag, at that point, we'll have the same problem: `prevProps.hasTags` will be `true`.

Reply:
In any project that uses Redux I'm always sad about using component `state` because we lose out on Redux debugging features. I also think it's nicer when things are all props over using component state, but maybe that's just me...

I can see this being more complex if you dispatched an action based on whether this post had any tags when the preview panel appeared, but that's instinctively what I'd prefer.

-----

Anyway, failing that, I think this could be better-named. Because it's similarly named to `hasTags` I thought it was just an assignment to `state` and was confused. Maybe:

```js
class MaybeTagsPanel extends Component {
	constructor( props ) {
		super( props );
		this.state = {
			showAddTagsPanel: ! props.hasTags,
		};
	}

	render() {
		if ( this.state.showAddTagsPanel ) {
			return <TagsPanel />;
		}

		// This post had tags before this component was loaded, so
		// we won't show a tag suggestion panel.
		return null;
	}
}
```

How about that?

Reply:
Hope this helps 0bcaa57